internal/server/data/access_key.go

```diff
@@ -174,25 +174,52 @@ func ListAccessKeys(tx ReadTxn, opts ListAccessKeyOptions) ([]models.AccessKey,
 type GetAccessKeysOptions struct {
 	ByID    uid.ID
 	ByKeyID string
+	ByName  string
 }
 
-// GetAccessKey using the keyID. Note that the keyID is globally unique, so
+// GetAccessKeyByKeyID using the keyID. Note that the keyID is globally unique,
+// and this function is meant to be called as part of autentication so
 // this query is not scoped by an organization_id.
-func GetAccessKey(tx ReadTxn, opts GetAccessKeysOptions) (*models.AccessKey, error) {
-	if opts.ByID == 0 && len(opts.ByKeyID) == 0 {
-		return nil, fmt.Errorf("GetAccessKey must supply either id or key_id")
+func GetAccessKeyByKeyID(tx ReadTxn, keyID string) (*models.AccessKey, error) {
+	if len(keyID) == 0 {
+		return nil, fmt.Errorf("GetAccessKeyByKeyID must key_id")
 	}
+
 	accessKey := &accessKeyTable{}
 	query := querybuilder.New("SELECT")
 	query.B(columnsForSelect(accessKey))
 	query.B("FROM")
 	query.B(accessKey.Table())
 	query.B("WHERE deleted_at is null")
+	query.B("AND key_id = ?", keyID)
+
+	err := tx.QueryRow(query.String(), query.Args...).Scan(accessKey.ScanFields()...)
+	if err != nil {
+		return nil, handleReadError(err)
+	}
+	return (*models.AccessKey)(accessKey), nil
+}
+
+// GetAccessKey by any unique field. This must be scoped by organizationID as it's callable by users.
+func GetAccessKey(tx ReadTxn, opts GetAccessKeysOptions) (*models.AccessKey, error) {
+	if opts.ByID == 0 && len(opts.ByKeyID) == 0 && len(opts.ByName) == 0 {
```

Comment:
I think we should probably remove `ByKeyID` from `GetAccessKeysOptions` (and the two places we reference that below here), so that there's not two ways to get the access key by KeyID. Otherwise it's possible someone uses the wrong function.